apiserver/uniter/uniter.go

```diff
@@ -433,7 +433,9 @@ func (u *UniterAPIV3) GetPrincipal(args params.Entities) (params.StringBoolResul
 			if err == nil {
 				principal, ok := unit.PrincipalName()
 				if principal != "" {
-					result.Results[i].Result = names.NewUnitTag(principal).String()
+					result.Results[i].Result = principal
+				} else {
+					result.Results[i].Result = unit.Name()
```

Comment:
can you please leave this line out? it's pretty confusing to me at least, as only a subordinate unit should have a principal unit

if we leave this out, then we can trivially know that a unit is subordinate by checking for the presence of the $JUJU_PRINCIPAL_UNIT env var. If that doesn't exist, we still have $JUJU_UNIT_NAME.